### ISSUE TITLE
144hz support?

### ISSUE THREAD

**Issue by JACHUPOL** [OneLiberty/moonlight-chrome-tizen#29]
As title says, would be a nice addition.

**Comment by OneLiberty** [comment]
As for now I don't have a way to monitor rendered FPS, I don't want to allow it 

Plus, it is very likely that most GPU won't be able to handle it at anything higher than 1080p. 

Planned but not "soon"

**Comment by JACHUPOL** [comment]
Thanks for reply, just so you know I have a 3070ti laptop and I'm streaming at 1440p 60hz at 40mbps so maybe it can do it at higher bitrate idk

**Comment by OneLiberty** [comment]
PC side is not the problem. 
I'm talking TV side. TVs usually have really bad CPU/GPU...

**Comment by AlexPresso** [comment]
Probably it would be possible with AV1 codec

**Comment by OneLiberty** [comment]
Well, isn't AV1 way more CPU intensive ?

**Comment by charnet3d** [comment]
I did a quick test with the higher FPS options 120/144 by adding them to the UI and it does not run well even on a 2024 premium TV, it keeps skipping frames and starts feeling like it's only running at 30FPS.

Heck even at 4K 60 I noticed some sessions I started I got consistent frameskipping as if it's only rendering 50 or 55 fps. At one point I was frustrated and kept testing different FPS limits in game and it only became smooth once the limit was 50, confirming the actual FPS on the stream. One workaround I found that can fix this is to set a lower resolution like 1440p and then back to 4K and it seems to work fine the 2nd time.

**Comment by JACHUPOL** [comment]
Can you share the version with the 120/144 hz options? Also are you sure they do correctly apply since on my TV 144hz is applied in game mode for example when plugged in via HDMI, I'm not sure what framerate is the TV running at normally. As I'm streaming to the TV in 1440p, 120hz should work just fine.

**Comment by charnet3d** [comment]
While the UI does set the options for 120/144 FPS, and I think does send them to the server, I'm not sure if there is something else that tells the app that it needs to run at a specific Hz, maybe the client tries to render those frames but the screen isn't set to an appropriate refresh rate.
Or there could even be too many dropped frame due to the GPU not being able to decode those high frames. I did see that [120hz is supported among the options that can be decoded](https://developer.samsung.com/smarttv/develop/specifications/media-specifications/2024-tv-video-specifications.html#24TV_PREMIUM4-(UHD)) so who knows. 

I'm still new to Tizen OS and JS/Wasm so still discovering things myself.

If you want to compile yourself, this git patch enables the 2 extra FPS options, increases the max bitrate to 120mbps (for 4k 144) and sets recommended values for the bitrate for each of the new resolution/fps combinations.

[0001-Blindly-enabling-120-144FPS-options.patch](https://github.com/user-attachments/files/17092187/0001-Blindly-enabling-120-144FPS-options.patch)

(should be applied after [the current latest commit](https://github.com/OneLiberty/moonlight-chrome-tizen/commit/4f0e8aba88f171b212361d86ca30c2583d860ae4) to avoid merge conflicts)

The compiled wgt file if you just want to test:

[Moonlight_120_144fps_for_testing.zip](https://github.com/user-attachments/files/17092198/Moonlight_120_144fps_for_testing.zip)

**Comment by Lyqed** [comment]
> While the UI does set the options for 120/144 FPS, and I think does send them to the server, I'm not sure if there is something else that tells the app that it needs to run at a specific Hz, maybe the client tries to render those frames but the screen isn't set to an appropriate refresh rate. Or there could even be too many dropped frame due to the GPU not being able to decode those high frames. I did see that [120hz is supported among the options that can be decoded](https://developer.samsung.com/smarttv/develop/specifications/media-specifications/2024-tv-video-specifications.html#24TV_PREMIUM4-(UHD)) so who knows.
> 
> I'm still new to Tizen OS and JS/Wasm so still discovering things myself.
> 
> If you want to compile yourself, this git patch enables the 2 extra FPS options, increases the max bitrate to 120mbps (for 4k 144) and sets recommended values for the bitrate for each of the new resolution/fps combinations.
> 
> [0001-Blindly-enabling-120-144FPS-options.patch](https://github.com/user-attachments/files/17092187/0001-Blindly-enabling-120-144FPS-options.patch)
> 
> (should be applied after [the current latest commit](https://github.com/OneLiberty/moonlight-chrome-tizen/commit/4f0e8aba88f171b212361d86ca30c2583d860ae4) to avoid merge conflicts)
> 
> The compiled wgt file if you just want to test:
> 
> [Moonlight_120_144fps_for_testing.zip](https://github.com/user-attachments/files/17092198/Moonlight_120_144fps_for_testing.zip)

@charnet3d 

Hi, trying to get it to work but after doing the install command for your .wgt file I get a certificate error. I assume I need to apply the patch and compile it myself, then sign it in the tizen ide?

**Comment by evildarkn3ss** [comment]
> > While the UI does set the options for 120/144 FPS, and I think does send them to the server, I'm not sure if there is something else that tells the app that it needs to run at a specific Hz, maybe the client tries to render those frames but the screen isn't set to an appropriate refresh rate. Or there could even be too many dropped frame due to the GPU not being able to decode those high frames. I did see that [120hz is supported among the options that can be decoded](https://developer.samsung.com/smarttv/develop/specifications/media-specifications/2024-tv-video-specifications.html#24TV_PREMIUM4-(UHD)) so who knows.
> > I'm still new to Tizen OS and JS/Wasm so still discovering things myself.
> > If you want to compile yourself, this git patch enables the 2 extra FPS options, increases the max bitrate to 120mbps (for 4k 144) and sets recommended values for the bitrate for each of the new resolution/fps combinations.
> > [0001-Blindly-enabling-120-144FPS-options.patch](https://github.com/user-attachments/files/17092187/0001-Blindly-enabling-120-144FPS-options.patch)
> > (should be applied after [the current latest commit](https://github.com/OneLiberty/moonlight-chrome-tizen/commit/4f0e8aba88f171b212361d86ca30c2583d860ae4) to avoid merge conflicts)
> > The compiled wgt file if you just want to test:
> > [Moonlight_120_144fps_for_testing.zip](https://github.com/user-attachments/files/17092198/Moonlight_120_144fps_for_testing.zip)
> 
> @charnet3d
> 
> Hi, trying to get it to work but after doing the install command for your .wgt file I get a certificate error. I assume I need to apply the patch and compile it myself, then sign it in the tizen ide?

I'm indeed struggling with the certificate error. PS. Thank you for this Tizen solution, had nasty issues with the google tv streamer regarding audio delays, this has zero audio delay. All i want to try now is the 120hz options but i'm stuck..

**Comment by charnet3d** [comment]
@Lyqed that's right. For the compilation I'll refer you to this [Comment](https://github.com/OneLiberty/moonlight-chrome-tizen/issues/37#issuecomment-2118982596) as it can be a bit lengthy if it's the first time you're doing it.

I should reiterate though that the 120+ option doesn't seem to be working correctly yet, it skips frames a lot and feels like 30fps, even with a more recent 2024 TV.

**Comment by glink00** [comment]
> @Lyqed that's right. For the compilation I'll refer you to this [Comment](https://github.com/OneLiberty/moonlight-chrome-tizen/issues/37#issuecomment-2118982596) as it can be a bit lengthy if it's the first time you're doing it.
> 
> I should reiterate though that the 120+ option doesn't seem to be working correctly yet, it skips frames a lot and feels like 30fps, even with a more recent 2024 TV.

I'd like to double tap on this. Unfortunately, for me it also feels as if I was playing on 30fps even with 60fps settings (samsung Oled S95B). The chromeOS moonlight app which is being used as base here does not have an overlay to easily validate the output fps. Maybe that would be a good starting point to help us gather reports from others. 

I've been searching for minipc's to connect to the TV that support 4k@120@4:4:4 (hdr) and it's actually not an easy find. With this information, my hunch is that it is likely that the TV decoders aren't capable of this since most movies and sports(most common TV use cases)  are streamed at max 60fps, so why would manufacturers bother. I wish I am wrong though :D

**Comment by charnet3d** [comment]
Not sure how it works on older hardware but on my S90D it's fine on 60fps. Maybe something is getting constrained and you drop some frames.
Samsung's spec table does indicate that the hardware should be capable of decoding 120fps content in the supported models, but whether it's on the native app or also in a chrome widget not sure, maybe it's a software limitation with the ChromeOS part.
And yeah ofc a way to diagnose FPS and decoding issues would greatly help in getting issues like these fixed.

For the MiniPC thing maybe try building a mini-ITX PC with a 10 series or newer nvidia card, won't be dirt cheap but should give you peace of mind when it comes to playing content on the tv and even light gaming. That's what I'm doing myself I have a 3080ti ITX PC I play with occasionally but for the latest AAA games I stream 4k144 to my workstation 4090.
